{"text": "2024-08-19 00:45:05 | INFO    | __main__:stash:131 | run - {'image': '20231121', 'version': '0.10.1.post2', 'role': 'EpicPlayer', 'headless': True}\n", "record": {"elapsed": {"repr": "0:00:01.143457", "seconds": 1.143457}, "exception": null, "extra": {"image": "20231121", "version": "0.10.1.post2", "role": "EpicPlayer", "headless": true}, "file": {"name": "claim.py", "path": "/home/runner/work/epic-game/epic-game/epic/src/claim.py"}, "function": "stash", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 131, "message": "run", "module": "claim", "name": "__main__", "process": {"id": 3675, "name": "MainProcess"}, "thread": {"id": 140591670987648, "name": "MainThread"}, "time": {"repr": "2024-08-19 00:45:05.940574+08:00", "timestamp": 1723999505.940574}}}
{"text": "2024-08-19 00:45:10 | WARNING    | epic_games.agent:get_order_history:467 | Failed to get order history, cookie may have expired - {}\n", "record": {"elapsed": {"repr": "0:00:05.551163", "seconds": 5.551163}, "exception": null, "extra": {}, "file": {"name": "agent.py", "path": "/home/runner/work/epic-game/epic-game/epic/src/epic_games/agent.py"}, "function": "get_order_history", "level": {"icon": "‚ö†Ô∏è", "name": "WARNING", "no": 30}, "line": 467, "message": "Failed to get order history, cookie may have expired", "module": "agent", "name": "epic_games.agent", "process": {"id": 3675, "name": "MainProcess"}, "thread": {"id": 140591670987648, "name": "MainThread"}, "time": {"repr": "2024-08-19 00:45:10.348280+08:00", "timestamp": 1723999510.34828}}}
{"text": "2024-08-19 00:45:11 | DEBUG    | __main__:create_tasks:66 | Put task - {'title': 'World of Warships ‚Äî Starter Pack: Albany', 'url': 'https://store.epicgames.com/en-US/p/world-of-warships'}\n", "record": {"elapsed": {"repr": "0:00:06.380742", "seconds": 6.380742}, "exception": null, "extra": {"title": "World of Warships ‚Äî Starter Pack: Albany", "url": "https://store.epicgames.com/en-US/p/world-of-warships"}, "file": {"name": "claim.py", "path": "/home/runner/work/epic-game/epic-game/epic/src/claim.py"}, "function": "create_tasks", "level": {"icon": "üêû", "name": "DEBUG", "no": 10}, "line": 66, "message": "Put task", "module": "claim", "name": "__main__", "process": {"id": 3675, "name": "MainProcess"}, "thread": {"id": 140591670987648, "name": "MainThread"}, "time": {"repr": "2024-08-19 00:45:11.177859+08:00", "timestamp": 1723999511.177859}}}
{"text": "2024-08-19 00:45:11 | DEBUG    | __main__:create_tasks:66 | Put task - {'title': \"Death's Gambit - Afterlife\", 'url': 'https://store.epicgames.com/en-US/p/deaths-gambit-afterlife-b5f411'}\n", "record": {"elapsed": {"repr": "0:00:06.381213", "seconds": 6.381213}, "exception": null, "extra": {"title": "Death's Gambit - Afterlife", "url": "https://store.epicgames.com/en-US/p/deaths-gambit-afterlife-b5f411"}, "file": {"name": "claim.py", "path": "/home/runner/work/epic-game/epic-game/epic/src/claim.py"}, "function": "create_tasks", "level": {"icon": "üêû", "name": "DEBUG", "no": 10}, "line": 66, "message": "Put task", "module": "claim", "name": "__main__", "process": {"id": 3675, "name": "MainProcess"}, "thread": {"id": 140591670987648, "name": "MainThread"}, "time": {"repr": "2024-08-19 00:45:11.178330+08:00", "timestamp": 1723999511.17833}}}
{"text": "2024-08-19 00:45:15 | INFO    | epic_games.agent:claim_weekly_games:294 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/world-of-warships'}\n", "record": {"elapsed": {"repr": "0:00:10.968048", "seconds": 10.968048}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/world-of-warships"}, "file": {"name": "agent.py", "path": "/home/runner/work/epic-game/epic-game/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 294, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3675, "name": "MainProcess"}, "thread": {"id": 140591670987648, "name": "MainThread"}, "time": {"repr": "2024-08-19 00:45:15.765165+08:00", "timestamp": 1723999515.765165}}}
{"text": "2024-08-19 00:45:28 | INFO    | epic_games.agent:claim_weekly_games:294 | claim_weekly_games - {'action': 'go to store', 'url': 'https://store.epicgames.com/en-US/p/deaths-gambit-afterlife-b5f411'}\n", "record": {"elapsed": {"repr": "0:00:24.144737", "seconds": 24.144737}, "exception": null, "extra": {"action": "go to store", "url": "https://store.epicgames.com/en-US/p/deaths-gambit-afterlife-b5f411"}, "file": {"name": "agent.py", "path": "/home/runner/work/epic-game/epic-game/epic/src/epic_games/agent.py"}, "function": "claim_weekly_games", "level": {"icon": "‚ÑπÔ∏è", "name": "INFO", "no": 20}, "line": 294, "message": "claim_weekly_games", "module": "agent", "name": "epic_games.agent", "process": {"id": 3675, "name": "MainProcess"}, "thread": {"id": 140591670987648, "name": "MainThread"}, "time": {"repr": "2024-08-19 00:45:28.941854+08:00", "timestamp": 1723999528.941854}}}
{"text": "2024-08-19 00:45:42 | ERROR    | __main__:run:158 | An error has been caught in function 'run', process 'MainProcess' (3675), thread 'MainThread' (140591670987648): - {}\nTraceback (most recent call last):\n  File \"/home/runner/work/epic-game/epic-game/epic/src/claim.py\", line 162, in <module>\n    asyncio.run(run())\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/asyncio/runners.py\", line 44, in run\n    return loop.run_until_complete(main)\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/asyncio/base_events.py\", line 636, in run_until_complete\n    self.run_forever()\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/asyncio/base_events.py\", line 603, in run_forever\n    self._run_once()\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/asyncio/base_events.py\", line 1909, in _run_once\n    handle._run()\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/asyncio/events.py\", line 80, in _run\n    self._context.run(self._callback, *self._args)\n> File \"/home/runner/work/epic-game/epic-game/epic/src/claim.py\", line 158, in run\n    await agent.stash()\n  File \"/home/runner/work/epic-game/epic-game/epic/src/claim.py\", line 151, in stash\n    await self.claim_epic_games(context)\n  File \"/home/runner/work/epic-game/epic-game/epic/src/claim.py\", line 122, in claim_epic_games\n    await epic.claim_weekly_games(page, single_promotions)\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/tenacity/asyncio/__init__.py\", line 189, in async_wrapped\n    return await copy(fn, *args, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/tenacity/asyncio/__init__.py\", line 111, in __call__\n    do = await self.iter(retry_state=retry_state)\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/tenacity/asyncio/__init__.py\", line 153, in iter\n    result = await action(retry_state)\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/tenacity/_utils.py\", line 99, in inner\n    return call(*args, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/tenacity/__init__.py\", line 398, in <lambda>\n    self._add_action_func(lambda rs: rs.outcome.result())\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/concurrent/futures/_base.py\", line 451, in result\n    return self.__get_result()\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/concurrent/futures/_base.py\", line 403, in __get_result\n    raise self._exception\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/tenacity/asyncio/__init__.py\", line 114, in __call__\n    result = await fn(*args, **kwargs)\n  File \"/home/runner/work/epic-game/epic-game/epic/src/epic_games/agent.py\", line 310, in claim_weekly_games\n    await expect(cta_btn).to_have_text(\"View In Cart\")\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/playwright/async_api/_generated.py\", line 19170, in to_have_text\n    await self._impl_obj.to_have_text(\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/playwright/_impl/_assertions.py\", line 481, in to_have_text\n    await self._expect_impl(\n  File \"/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/playwright/_impl/_assertions.py\", line 74, in _expect_impl\n    raise AssertionError(\nAssertionError: Locator expected to have text 'View In Cart'\nActual value: <element(s) not found> \nCall log:\nLocatorAssertions.to_have_text with timeout 5000ms\n  - waiting for locator(\"//aside//button[@data-testid='add-to-cart-cta-button']\")\n\n", "record": {"elapsed": {"repr": "0:00:37.536997", "seconds": 37.536997}, "exception": {"type": "AssertionError", "value": "Locator expected to have text 'View In Cart'\nActual value: <element(s) not found> \nCall log:\nLocatorAssertions.to_have_text with timeout 5000ms\n  - waiting for locator(\"//aside//button[@data-testid='add-to-cart-cta-button']\")\n", "traceback": true}, "extra": {}, "file": {"name": "claim.py", "path": "/home/runner/work/epic-game/epic-game/epic/src/claim.py"}, "function": "run", "level": {"icon": "‚ùå", "name": "ERROR", "no": 40}, "line": 158, "message": "An error has been caught in function 'run', process 'MainProcess' (3675), thread 'MainThread' (140591670987648):", "module": "claim", "name": "__main__", "process": {"id": 3675, "name": "MainProcess"}, "thread": {"id": 140591670987648, "name": "MainThread"}, "time": {"repr": "2024-08-19 00:45:42.334114+08:00", "timestamp": 1723999542.334114}}}
